### PR TITLE
Enabled "No Mouse Fridays" in development mode.

### LINKED ISSUE #10260: Consider No Mouse Fridays in the Application
**Is your feature request related to a problem? Please describe.**

Consider having "No Mouse Fridays" as part of our accessibility initiative.

**Describe the solution you'd like**

The simplest way to do this would be to include Marcy Sutton's package [No Mouse Days](https://github.com/marcysutton/no-mouse-days) when running in development mode.

**Describe alternatives you've considered**

This could be done with a browser extension that allows you to modify CSS for a page, but that requires work which means friction.

**Additional context**

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] DX for Accessibility

## Description

This adds Marcy Sutton's [No Mouse Days](https://github.com/marcysutton/no-mouse-days) package when running in development mode on Fridays so that mouse activity is disabled.

## Related Tickets & Documents

- Closes #10260
- [Webpack Entry Points](https://webpack.js.org/concepts/entry-points/) documentation
- [Live Stream](https://www.twitch.tv/videos/783457919) of the work for this pull request

### Generated Webpack Configuration for Entry Points Before

```javascript
"entry": {
    "actionsPanel": "/Users/john_doe/dev/forem/app/javascript/packs/actionsPanel.js",
    "admin": "/Users/john_doe/dev/forem/app/javascript/packs/admin.js",
    "analyticsArticle": "/Users/john_doe/dev/forem/app/javascript/packs/analyticsArticle.js",
    "analyticsDashboard": "/Users/john_doe/dev/forem/app/javascript/packs/analyticsDashboard.js",
    "articleForm": "/Users/john_doe/dev/forem/app/javascript/packs/articleForm.jsx",
    "articleModerationTools": "/Users/john_doe/dev/forem/app/javascript/packs/articleModerationTools.js",
    "articlePage": "/Users/john_doe/dev/forem/app/javascript/packs/articlePage.jsx",
    "Chat": "/Users/john_doe/dev/forem/app/javascript/packs/Chat.jsx",
    "clipboardCopy": "/Users/john_doe/dev/forem/app/javascript/packs/clipboardCopy.js",
    "colorPicker": "/Users/john_doe/dev/forem/app/javascript/packs/colorPicker.js",
    "colorPreview": "/Users/john_doe/dev/forem/app/javascript/packs/colorPreview.js",
    "commentModPage": "/Users/john_doe/dev/forem/app/javascript/packs/commentModPage.js",
    "contentDisplayPolicy": "/Users/john_doe/dev/forem/app/javascript/packs/contentDisplayPolicy.js",
    "flagUserModal": "/Users/john_doe/dev/forem/app/javascript/packs/flagUserModal.jsx",
    "githubRepos": "/Users/john_doe/dev/forem/app/javascript/packs/githubRepos.jsx",
    "homePage": "/Users/john_doe/dev/forem/app/javascript/packs/homePage.jsx",
    "homePageFeed": "/Users/john_doe/dev/forem/app/javascript/packs/homePageFeed.jsx",
    "listingDashboard": "/Users/john_doe/dev/forem/app/javascript/packs/listingDashboard.jsx",
    "listingForm": "/Users/john_doe/dev/forem/app/javascript/packs/listingForm.jsx",
    "listings": "/Users/john_doe/dev/forem/app/javascript/packs/listings.jsx",
    "modCenter": "/Users/john_doe/dev/forem/app/javascript/packs/modCenter.jsx",
    "notificationSubscriptionHandler": "/Users/john_doe/dev/forem/app/javascript/packs/notificationSubscriptionHandler.js",
    "Onboarding": "/Users/john_doe/dev/forem/app/javascript/packs/Onboarding.jsx",
    "onboardingRedirectCheck": "/Users/john_doe/dev/forem/app/javascript/packs/onboardingRedirectCheck.jsx",
    "orgCreditsSelector": "/Users/john_doe/dev/forem/app/javascript/packs/orgCreditsSelector.js",
    "profileDropdown": "/Users/john_doe/dev/forem/app/javascript/packs/profileDropdown.js",
    "readingList": "/Users/john_doe/dev/forem/app/javascript/packs/readingList.jsx",
    "responseTemplates": "/Users/john_doe/dev/forem/app/javascript/packs/responseTemplates.js",
    "Search": "/Users/john_doe/dev/forem/app/javascript/packs/Search.jsx",
    "sidebarWidget": "/Users/john_doe/dev/forem/app/javascript/packs/sidebarWidget.jsx",
    "stickySaveFooter": "/Users/john_doe/dev/forem/app/javascript/packs/stickySaveFooter.js",
    "validateFileInputs": "/Users/john_doe/dev/forem/app/javascript/packs/validateFileInputs.js",
    "videoChat": "/Users/john_doe/dev/forem/app/javascript/packs/videoChat.jsx",
    "webShare": "/Users/john_doe/dev/forem/app/javascript/packs/webShare.js"
  },
```

### Generated Webpack Configuration for Entry Points After

```javascript
  "entry": {
    "actionsPanel": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/actionsPanel.js"
    ],
    "admin": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/admin.js"
    ],
    "analyticsArticle": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/analyticsArticle.js"
    ],
    "analyticsDashboard": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/analyticsDashboard.js"
    ],
    "articleForm": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/articleForm.jsx"
    ],
    "articleModerationTools": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/articleModerationTools.js"
    ],
    "articlePage": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/articlePage.jsx"
    ],
    "Chat": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/Chat.jsx"
    ],
    "clipboardCopy": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/clipboardCopy.js"
    ],
    "colorPicker": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/colorPicker.js"
    ],
    "colorPreview": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/colorPreview.js"
    ],
    "commentModPage": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/commentModPage.js"
    ],
    "contentDisplayPolicy": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/contentDisplayPolicy.js"
    ],
    "flagUserModal": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/flagUserModal.jsx"
    ],
    "githubRepos": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/githubRepos.jsx"
    ],
    "homePage": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/homePage.jsx"
    ],
    "homePageFeed": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/homePageFeed.jsx"
    ],
    "listingDashboard": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/listingDashboard.jsx"
    ],
    "listingForm": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/listingForm.jsx"
    ],
    "listings": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/listings.jsx"
    ],
    "modCenter": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/modCenter.jsx"
    ],
    "notificationSubscriptionHandler": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/notificationSubscriptionHandler.js"
    ],
    "Onboarding": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/Onboarding.jsx"
    ],
    "onboardingRedirectCheck": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/onboardingRedirectCheck.jsx"
    ],
    "orgCreditsSelector": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/orgCreditsSelector.js"
    ],
    "profileDropdown": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/profileDropdown.js"
    ],
    "readingList": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/readingList.jsx"
    ],
    "responseTemplates": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/responseTemplates.js"
    ],
    "Search": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/Search.jsx"
    ],
    "sidebarWidget": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/sidebarWidget.jsx"
    ],
    "stickySaveFooter": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/stickySaveFooter.js"
    ],
    "validateFileInputs": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/validateFileInputs.js"
    ],
    "videoChat": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/videoChat.jsx"
    ],
    "webShare": [
      "./app/javascript/noMouseDays.js",
      "/Users/john_doe/dev/forem/app/javascript/packs/webShare.js"
    ]
  },
```

To test:
1. If this pull request is being reviewed on a Friday, there is nothing to change when running the pull request locally. If it is being reviewed on any other day than a Friday change this line in the noMouseDays.js file from `dayOfWeek === 5` to `dayOfWeek === n`, where n is the number representing the day of the week.
1. A snackbar message appears indicating it is no mouse Fridays
1. Notice that the mouse cursor is no longer visible and nothing in the application can be clicked on.

no-mouse-message

## Added tests?

- [ ] Yes
- [x] No, and this is why: This is a webpack enabled development mode only feature for disabling mouse navigation on Fridays.
- [ ] I need help with writing tests

## Added to documentation?

- [ ] Docs.forem.com
- [ ] README
- [x] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

A mouse rolling off of a pop bottle

### CODE REVIEW

File: app/javascript/noMouseDays.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const date = new Date();
+const dayOfWeek = date.getDay();
+
+// We only want to disable mouse usage
+// on Fridays.
+if (dayOfWeek === 5) {
+  import('no-mouse-days');
```
Comment (nickytonline) [inline app/javascript/noMouseDays.js:6]: The [dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) can just be imported. There is no data returned so just let the Promise execute.

**Comment by citizen428** [comment]
How do I opt-out of this? I already use a very keyboard drive workflow with Vimium, but I'm not keen on spending time figuring out why something isn't working before I remember that it's Friday...

File: package.json [hunk 2]
```diff
@@ -123,6 +123,7 @@
     "jsdom": "^16.4.0",
     "lint-staged": "^10.4.2",
     "markdown-loader": "^6.0.0",
+    "no-mouse-days": "^0.0.3",
```
Comment (nickytonline) [inline package.json:126]: The package is required only for development mode, so it can reside in `devDependencies`.

File: config/webpack/environment.js [hunk 3]
```diff
@@ -67,4 +67,16 @@ if (process.env.HONEYBADGER_API_KEY && process.env.ASSETS_URL) {
   );
 }
 
+if (process.env.NODE_ENV === 'development') {
+  // This disables mouse usage in the application in development mode only
+  // to ensure the application is accessible via keyboard navigation.
+  // See https://github.com/forem/forem/issues/10260
+  const noMouseDaysScript = './app/javascript/noMouseDays.js';
+
+  Object.keys(environment.entry).forEach((key) => {
```
Comment (nickytonline) [inline config/webpack/environment.js:76]: This alters the webpack configuration for entry points to include the no mouse days script for each entry point.

**Comment by nickytonline** [comment]
> How do I opt-out of this? I already use a very keyboard driven workflow with Vimium, but I'm not keen on spending time figuring out why something isn't working before I remember that it's Friday...

I still need to add an indication on the screen that the mouse is deactivated, i.e. Friday. In terms of opting out, this has not been considered as it defeats the purpose of a "no mouse Friday".

**Comment by citizen428** [comment]
> In terms of opting out in development mode on Fridays, this has not been considered as it defeats the purpose of a "no mouse Friday".

I like the idea in general, but I just feel like maybe some Fridays I really want to finish something (last Friday of a cycle for example) and would prefer to work in . Maybe we can add an env variable (`NO_MOUSE_FRIDAYS`) that’s true by default so on Friday's where one doesn't want this behavior it can explicitly turned off.

**Comment by ludwiczakpawel** [comment]
I like the idea of encouraging people to only use keyboard to improve awareness around accessibility and to improve that way of navigating the site in general. But I'm **very against** adding any library that would force me to do so.. I can't imagine myself not using mouse on Fridays, especially when there's an emergency or when I'm simply in a "_I'm building something_" mode... If I'd have to manually opt-out from this, I think that would be pretty rough development experience. Not even mention external contributors who may not be aware of "No Mouse Fridays" idea in general. Even with the _Snackbar notification_ showing info about it, I think ppl would be pretty confused and annoyed (maybe even pissed off is some extreme cases). 

If we want to improve keyboard accessibility awareness within Forem, we should probably do it on a company level in some non-distruptive way as well as by introducing some processes around building features (_"hey, you need to add tabindex and aria-label, etc. etc. before making a PR"_ -kind of a thing).

Therefore if anyone wants to use keyboard only on a given day (or even period of time), they can simply force themselves to do so - I don't think it requires a library.

**TL;DR:** Please no 🙏.

**Review by mstruve — COMMENTED** [review]
I'm a little worried that this might catch some people by surprise who may not want it and they get stuck no knowing how to disable it. I could 100% see me running into the code on Friday trying to test something really quick and getting extremely confused. 

Could we disable this by default and somehow alert people to the option so they can choose to turn it on if they want?

**Comment by nickytonline** [comment]
@citizen428 @mstruve @ludwiczakpawel, I like @citizen428’s suggestion of making this opt-out, but I hadn’t considered external contributors (who do not work in frontend) potentially getting annoyed, or folks on deadlines on a Friday, so opt-in seems to make more sense. It kills the spirit of the idea a bit, but these are valid points.

Would folks be open to the boot up of the app in development mode asking if they want to enable "No Mouse Fridays"? That way, it’s opt-out by default.

**Comment by msarit** [comment]
I agree with making users opt-in to NoMouseFridays. Perhaps we could display a banner on the homepage on Fridays, alerting users to the option of turning NoMouseFridays on?

**Comment by ludwiczakpawel** [comment]
Sorry if I killed the spirit or all the fun part about this PR :d But I'll be keep pushing back on this one. I still think this will be probably annoying even if its a prompt on the boot up that needs to be clicked. Every day, I try to review at least several PRs, every time spinning up bin/setup and bin/startup so I do wish I wouldn't have to take any extra step slowing down the process. 

I still think you can simply not use your mouse if you feel like it. The benefits over proposed solution would be:

1. You don't need extra library for not touching your mouse :D 
2. You can do it on ANY day you want, not only Fridays.
3. You can NOT do it if you don't have to.

FWIW, I think we could have similar results if we set up company-wide notification in Google Calendar and/or Slack to prompt people about not using mouse every Friday...

**Comment by rhymes** [comment]
I agree with the concerns expressed by others as I think this should definitely be an opt in, if we decide to move on with it. 

Third party contributors casually helping out or testing code on Friday might be taken aback, especially as there's no notice or banner.

I'd rather this became a documentation page first, something like "this is what you have to do to turn on no-mouse-mode" and then we can, as a company or individually, join in on fridays regularly or start with a cool down sprint a day or two to see how we feel about it :)

Thanks for kickstarting the conversation!

**Comment by citizen428** [comment]
@nickytonline Should we close this PR and revisit the idea in an RFC?

**Comment by citizen428** [comment]
@nickytonline As this has seen no activity for a bit I'll close it now. We can revisit the idea via an RFC if you're still interested in it.

**Comment by nickytonline** [comment]
Sounds good @citizen428.
